nixos/tests/cockroachdb.nix

```diff
@@ -110,17 +110,19 @@ in import ./make-test.nix ({ pkgs, ...} : {
   # there's otherwise no way to guarantee that node1 will start before the others try
   # to join it.
   testScript = ''
-    $node1->start;
-    $node1->waitForUnit("cockroachdb");
+    start_all()
+    node1.wait_for_unit("cockroachdb")
+    node2.wait_for_unit("cockroachdb")
+    node3.wait_for_unit("cockroachdb")
 
-    $node2->start;
-    $node2->waitForUnit("cockroachdb");
-
-    $node3->start;
-    $node3->waitForUnit("cockroachdb");
-
-    $node1->mustSucceed("cockroach sql --host=192.168.1.1 --insecure -e 'SHOW ALL CLUSTER SETTINGS' 2>&1");
-    $node1->mustSucceed("cockroach workload init bank 'postgresql://root\@192.168.1.1:26257?sslmode=disable'");
-    $node1->mustSucceed("cockroach workload run bank --duration=1m 'postgresql://root\@192.168.1.1:26257?sslmode=disable'");
+    node1.succeed(
+        "cockroach sql --host=192.168.1.1 --insecure -e 'SHOW ALL CLUSTER SETTINGS' 2>&1"
+    )
+    node1.succeed(
+        "cockroach workload init bank 'postgresql://root@192.168.1.1:26257?sslmode=disable'"
+    )
+    node1.succeed(
+        "cockroach workload run bank --duration=1m 'postgresql://root@192.168.1.1:26257?sslmode=disable'"
+    )
```

Comment:
As this is just one single testcase and you are not using individual command outputs, how about:

```python
    for node in node1, node2, node3:
        node.wait_for_unit("cockroachdb")

    node1.succeed(
        "cockroach sql --host=192.168.1.1 --insecure -e 'SHOW ALL CLUSTER SETTINGS' 2>&1",
        "cockroach workload init bank 'postgresql://root@192.168.1.1:26257?sslmode=disable'",
        "cockroach workload run bank --duration=1m 'postgresql://root@192.168.1.1:26257?sslmode=disable'"
    )
```

Reply:
I'm not sure. I get a lot of
```
node2 # [   47.361925] cockroach[743]: W191123 14:11:14.135297 110 vendor/google.golang.org/grpc/clientconn.go:1304  grpc: addrConn.createTransport failed to connect to {192.168.1.1:26257 0  <nil>}. Err :connection error: desc = "transport: Error while dialing cannot reuse client connection". Reconnecting...
```

Not sure when / if this converges.

Reply:
Can we somehow orchestrate the initial cluster join from the VM driver? Does this make any sense?